### PR TITLE
Implement preview client

### LINKED ISSUE #6: Preview API
Hi,

I am using this library to create a Nextcloud first Photos App for Android. What I am missing is access to image previews. If it is fine with you I will contribute the necessary implementation.

I intend to provide access to the same two endpoints that are used in the Java library as mentioned here:
https://help.nextcloud.com/t/getting-image-preview-with-android-library-or-via-webdav/75743

### PR DESCRIPTION
**Pull request by vauvenal5**
Allows to request previews and thumbnails from the Nextcloud server. Closes #6 

I did not bump the version number since I do not know in what steps you are going. I guess this would be 1.2.0? I can do it if you want.

### CODE REVIEW

**Comment by provokateurin** [comment]
Yes please bump the version to `1.2.0` and add a changelog and readme entry.

File: lib/src/preview/client.dart [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  String _getThumbnailUrl(String remotePath, int width, int height) {
+    if(!remotePath.startsWith('/')) {
+      remotePath = '/$remotePath';
+    }
+    return '$_baseUrl/index.php/apps/files/api/v1/thumbnail/$width/$height${Uri.encodeFull(remotePath)}';
+  }
+
+  /// fetch preview for [remotePath] with provided [width] and [height]
+  /// [width] and [height] default to 512
+  Future<Uint8List> getPreview(String remotePath, {int width = 512, int height = 512}) async {
```
Comment (provokateurin) [inline lib/src/preview/client.dart]: Are these default values also given in the java library? If not I would remove them so the user has to explicitly specify the dimensions.

File: lib/src/preview/client.dart [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  }
+
+  /// fetch thumbnail for [remotePath] with provided [width] and [height]
+  /// [width] and [height] default to 512
+  /// thumbnail will crop your image if you want resized images use preview
+  Future<ByteStream> getThumbnailStream(String remotePath, {int width = 512, int height = 512}) async {
+    final response = await _network.download('GET', _getThumbnailUrl(remotePath, width, height), [200]);
+    return response.stream;
+  }
+}
```
Comment (provokateurin) [inline lib/src/preview/client.dart]: Missing new line

**Review by provokateurin — CHANGES_REQUESTED** [review]
Could you also add a test to confirm this new feature works?

File: lib/src/preview/client.dart (hunk 1 above)
Comment (vauvenal5) [reply lib/src/preview/client.dart]: I removed the default values. Sizes are now mandatory.

File: lib/src/preview/client.dart (hunk 2 above)
Comment (vauvenal5) [reply lib/src/preview/client.dart]: Sorry not quite sure what you mean.

**Comment by vauvenal5** [comment]
I additionally added tests for preview, bumped the version and added to the changelog. I also renamed the test file to end in `_test.dart` according to the recommendations from the test package.

File: lib/src/preview/client.dart (hunk 2 above)
Comment (provokateurin) [reply lib/src/preview/client.dart]: Just add a final newline

Comment (vauvenal5) [reply lib/src/preview/client.dart]: done

**Review by provokateurin — APPROVED** [review]

**Comment by provokateurin** [comment]
Thanks for your contribution! Already published on https://pub.dev

**Pull request merged**